### ISSUE TITLE
Bug when using seqmerge mode

### ISSUE THREAD

**Issue by alper1976** [jtamames/SqueezeMeta#376]
Attached find some relevant log files and the run script

[mergelog.txt](https://github.com/jtamames/SqueezeMeta/files/7390210/mergelog.txt)
[syslog.txt](https://github.com/jtamames/SqueezeMeta/files/7390221/syslog.txt)
[slurm-4025637.out.txt](https://github.com/jtamames/SqueezeMeta/files/7390250/slurm-4025637.out.txt)
[mergedassemblies.seqmerge.runAmos.log.txt](https://github.com/jtamames/SqueezeMeta/files/7390323/mergedassemblies.seqmerge.runAmos.log.txt)

**Comment by alper1976** [comment]
Here the output from the last run. Seems like nucmer has issues.

[slurm-4274791.out.txt](https://github.com/jtamames/SqueezeMeta/files/7451648/slurm-4274791.out.txt)

**Comment by fpusan** [comment]
Hi!
What's the memory capacity of your nodes?
`MaxRSSTask 19338352K` may be due to a cap of 20 Gb for this task.

**Comment by alper1976** [comment]
Hi. Thanks for your help this far. I fixed the memory issues but I am now running into time issues. Any suggestions to optimize the scaffolding. I already removed contigs shorter than 500 bp and run it for 14 days on 12 threads.

**Comment by fpusan** [comment]
Hi,

Sorry for the late response. Time issues might be difficult to solve. Minimus2 is a large blocker in this regard, as it works well for a moderate amount of samples but scales poorly when the input data becomes large. We have not been able to find a direct replacement to it, so at some point you will have to use either a coassembly (if you have enough RAM) or analyze each sample separately with the sequential mode.

An intermediate road would be to divide your samples in groups. For example you could use [Mash](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0997-x) to calculate pairwise distances between your metagenomes, cluster them based on similarity, and then analyze each cluster of samples independently.

**Comment by blostein** [comment]
Hi @fpusan , 
Thanks for a great tool and being so responsive. I've used SqueezeMeta for smaller batches of ~15 samples. Now I have a very large project (485 samples, although this includes positive/negative controls & some low-read samples I could remove, >800GB, paired reads, sample files highly variable in read count but median 10 M reads) and I'm trying to strategize my approach. Initially I was planning on using seqmerge mode, but reading this issue I'm concerned merging time due to minimus2 will be a problem for me too, given the number of samples and the n-1 merges needed. I can remove some low-read count samples but while this would reduce the number of merges needed it would not reduce very much the size of the pooled samples. 

My goal is to compare general taxonomic and functional attributes across conditions but I'm also interested in bins.
 The following resources are available to me on a HPC, max walltime of 14 days: 
- 36 cores + 180GB RAM per node OR
- 36 cores + 1.5TB RAM per node 

Since I don't think Megahit is MPI enabled, I'm stuck with these per node limits. So far reading over the different issues on this, I think there are maybe a few different strategies: 
1. Assemble all samples individually in sequential mode, merge taxonomic and functional profiles in R following advice given in [153](https://github.com/jtamames/SqueezeMeta/issues/153). Maybe parallelize runs to increase speed. Accept lack of bins or use derep (or pyani) to try and dereplicate bins. 
2. Use MASH to cluster samples by similarity (maybe into groups of 25 or 50 samples each?), co-assemble each cluster group and either analyze separately or merge SQM objects in R, again use derep/pyani to depreplicate bins. 

Do you have any suggestions as to which would be better? or is there something else I should consider doing?

**Comment by fpusan** [comment]
Hi!

I'm a bit short of time but I'll try to give a quick answer. I would so for the first option you propose:

1) Run SqueezeMeta individually for each sample, only use metabat2 for binning
2) Combine the results as described in #153 
3) Get all the bins together and merge them into ANI > 95% clusters.
4) You can use drep to derreplicate, but by doing so you will lose some info about the accessory genome. I have a new preprint on how to tackle this, maybe it will be of interest? (https://www.biorxiv.org/content/10.1101/2022.03.25.485477v1)

Of course the option with MASH is also cool and sound, and it will assemble more low abundance taxa. But the first one is easier and scales better. If you end up comparing both, I'd love to know the results!

Also I noticed that you posted something about not being able to install SqueezeMeta with mamba, but now can't find the issue. Is the problem still happening?

**Comment by blostein** [comment]
Thank you! I will do as you suggest at first. I am a little concerned because there are some low read samples that will probably not have very good assembly to contigs on their own, but these samples probably shouldn't be analyzed anyway. I'll read your preprint, thanks very much! Depending how it goes, I may be back with more questions :grimacing:

Yes, sorry, I deleted the mamba comment - in order to install using mamba, I had to create a new base environment instead of using the base conda environment provided on my HP. Then I didn't realize that in my new base environment there was no mkl. Installing numpy to the new base environment got me mkl, so it was just me being dense.
